### PR TITLE
OCR Project 4

### PR DESCRIPTION
**Pull request by crm-ops**
Following comments from @fpatelm we are proposing the following changes :

- will delete crm-ops/sfdc-ocr-p4 cloned project
- we have forked  Project_DA_Java_EN_Come_to_the_Rescue_of_a_Java_Application
- AnalyticsCounter is now only a main method
- ISymptomReader interface is now implemented in ReadSymptomDataFromFile
- New class SymptomsCounter - does the map/reduce count operation
- New class SymptomsFileOutput - write the result ordered alphabetically
- Map object are used when necessary for code simplicity

### CODE REVIEW

File: Project02Eclipse/src/com/hemebiotech/analytics/AnalyticsCounter.java [hunk 1]
```diff
@@ -1,43 +1,63 @@
...
+		//first get input symptoms from file and store them in a temp list
+		ReadSymptomDataFromFile filereader = new ReadSymptomDataFromFile(path);
+		inputSymptoms.addAll(filereader.GetSymptoms());
+
+
+		// take the list and pass it to a map class method for count
+		SymptomsCounter symptomcounter = new SymptomsCounter(inputSymptoms);
+		symptomsreport = symptomcounter.getSymptomsCount();
+
+
```
Comment (fpatelm) [inline Project02Eclipse/src/com/hemebiotech/analytics/AnalyticsCounter.java]: remove empty lines

File: Project02Eclipse/src/com/hemebiotech/analytics/SymptomsCounter.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                results.put(str,1);
+            } else {
+
+                results.replace(str,results.get(str)+1);
+            }
+
+
+        }
+
+
```
Comment (fpatelm) [inline Project02Eclipse/src/com/hemebiotech/analytics/SymptomsCounter.java]: remove empty lines

File: Project02Eclipse/src/com/hemebiotech/analytics/SymptomsCounter.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        for (String str: inputList) {
+            //the map does not contain the key so we add the key and value 1
+            if(!results.containsKey(str)) {
+                results.put(str,1);
+            } else {
+
+                results.replace(str,results.get(str)+1);
+            }
+
```
Comment (fpatelm) [inline Project02Eclipse/src/com/hemebiotech/analytics/SymptomsCounter.java]: remove empty lines

**Review by fpatelm — CHANGES_REQUESTED** [review]
Just removing empty lines as per review. Good Job!

File: Project02Eclipse/src/com/hemebiotech/analytics/AnalyticsCounter.java (hunk 1 above)
Comment (crm-ops) [reply Project02Eclipse/src/com/hemebiotech/analytics/AnalyticsCounter.java]: Done on second commit

File: Project02Eclipse/src/com/hemebiotech/analytics/SymptomsCounter.java (hunk 2 above)
Comment (crm-ops) [reply Project02Eclipse/src/com/hemebiotech/analytics/SymptomsCounter.java]: done on second commit

File: Project02Eclipse/src/com/hemebiotech/analytics/SymptomsCounter.java (hunk 3 above)
Comment (crm-ops) [reply Project02Eclipse/src/com/hemebiotech/analytics/SymptomsCounter.java]: done on second commit

**Review by fpatelm — APPROVED** [review]

**Comment by fpatelm** [comment]
Approved

**Pull request merged**